include/stl2/detail/memory/uninitialized_copy.hpp

```diff
@@ -0,0 +1,72 @@
+// cmcstl2 - A concept-enabled C++ standard library
+//
+//  Copyright Casey Carter 2016
+//  Copyright Christopher Di Bella 2016
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/caseycarter/cmcstl2
+//
+#ifndef STL2_DETAIL_ALGORITHM_UNINITIALIZED_COPY_HPP
+#define STL2_DETAIL_ALGORITHM_UNINITIALIZED_COPY_HPP
+
+#include <new>
+#include <stl2/iterator.hpp>
+#include <stl2/detail/fwd.hpp>
+#include <stl2/detail/memory/destroy.hpp>
+#include <stl2/detail/memory/reference_to.hpp>
+#include <stl2/detail/tagged.hpp>
+
+STL2_OPEN_NAMESPACE {
+   ///////////////////////////////////////////////////////////////////////////
+   // uninitialized_copy [Extension]
+   //
+   template <InputIterator I, Sentinel<I> S, /*NoThrow*/ForwardIterator O>
```

Comment:
Ahh, so it's a semantic difference! I think I got hung up on `Destructible` having

```cpp
{ t.~T() } noexcept;
```

and thought the same could be applied to `NoThrowForwardIterator`. I'll fix in `stl2/proposals-master` as well. The sooner we can get axioms, the better (toying with making researching into it a PhD topic if I can come up with a cool enough idea and can brave the four years -- no promises though).